troposphere/asg-elb-dns.py

```diff
@@ -110,81 +94,121 @@ def generate_cloudformation_template():
         Default="300",
     ))
 
-    health_check_target = template.add_parameter(Parameter(
-        "LoadBalancerHealthCheckTarget",
-        Type="String",
-    ))
-
-    health_check_interval = template.add_parameter(Parameter(
-        "LoadBalancerHealthCheckInterval",
-        Type="String",
-    ))
-
-    health_check_timeout = template.add_parameter(Parameter(
-        "LoadBalancerHealthCheckTimeout",
-        Type="String",
-    ))
-
-    healthy_threshold = template.add_parameter(Parameter(
-        "LoadBalancerHealthyThreshold",
-        Type="String",
-    ))
-
-    unhealthy_threshold = template.add_parameter(Parameter(
-        "LoadBalancerUnHealthyThreshold",
-        Type="String",
-    ))
-
-    enable_connection_draining = template.add_parameter(Parameter(
-        "LoadBalancerEnableConnectionDraining",
-        Type="String",
-        Default="True",
-    ))
-
-    connection_draining_timeout = template.add_parameter(Parameter(
-        "LoadBalancerConnectionDrainingTimeout",
-        Type="String",
-        Default="30",
-    ))
-
-    launchconfigurationname = template.add_parameter(Parameter(
-        "LaunchConfigurationName",
-        Type="String",
-    ))
-
-    new_weight = template.add_parameter(Parameter(
-        "NewDnsWeight",
-        Type="String",
-        Default="100",
-    ))
-
-    loadbalancer = template.add_resource(elb.LoadBalancer(
-        "LoadBalancer",
-        ConnectionDrainingPolicy=elb.ConnectionDrainingPolicy(
-            Enabled=Ref(enable_connection_draining),
-            Timeout=Ref(connection_draining_timeout),
-        ),
-        Subnets=Ref(subnet),
-        HealthCheck=elb.HealthCheck(
-            Target=Ref(health_check_target),
-            HealthyThreshold=Ref(healthy_threshold),
-            UnhealthyThreshold=Ref(unhealthy_threshold),
-            Interval=Ref(health_check_interval),
-            Timeout=Ref(health_check_timeout),
-        ),
-        Listeners=[
-            elb.Listener(
-                LoadBalancerPort="80",
-                InstancePort="80",
-                Protocol="HTTP",
-                InstanceProtocol="HTTP",
+    if enable_elb:
```

Comment:
Probably the first suggestion is the one that makes more sense here. I'll try to apply it and see what I get! Thanks